### PR TITLE
Scripting changes to be able to use EKB with a tag

### PR DESCRIPTION
**Pull request by aliok**
- Scripting changes to be able to use EKB with a tag
- Also using a tag

### CODE REVIEW

**Comment by aliok** [comment]
/test 4.6-upstream-e2e-aws-ocp-46

```
level=fatal msg="failed to fetch Cluster: failed to generate asset \"Cluster\": failed to create cluster: failed to apply Terraform: failed to complete the change"
```

**Comment by aliok** [comment]
/retest

Infra issues

**Comment by aliok** [comment]
/test 4.6-upstream-e2e-aws-ocp-46

```
msg="Error: [ERR]: Error building changeset: InvalidChangeBatch: [Tried to create resource record set [name='api.ci-op-bc8icvkp-13ab6.origin-ci-int-aws.dev.rhcloud.com.', type='A'] but it already exists]"
level=error msg="\tstatus code: 400, request id: f2bf49d4-2163-4dc1-a777-44b94ff278b0"
```

**Comment by aliok** [comment]
/test 4.6-upstream-e2e-aws-ocp-46

Getting the same thing again and again
```
msg="Error: [ERR]: Error building changeset: InvalidChangeBatch: [Tried to create resource record set [name='api.ci-op-bc8icvkp-13ab6.origin-ci-int-aws.dev.rhcloud.com.', type='A'] but it already exists]"
```

**Comment by aliok** [comment]
/test 4.6-upstream-e2e-aws-ocp-46
/test 4.6-upstream-e2e-mesh-aws-ocp-46

Infra issues again
```
"failed to fetch Cluster: failed to generate asset \"Cluster\": failed to create cluster: failed to apply Terraform: failed to complete the change"
```

**Comment by aliok** [comment]
/test 4.6-upstream-e2e-aws-ocp-46

Infra

**Comment by aliok** [comment]
All green, improving commit history now

File: test/lib.bash [hunk 1]
```diff
@@ -185,20 +185,20 @@ function downstream_kitchensink_e2e_tests {
...
+  eventing_kafka_broker_test_images_version=$(versions.major_minor "${KNATIVE_EVENTING_KAFKA_BROKER_TEST_IMAGES_VERSION}")
 
   # mapping based on https://github.com/openshift/release/blob/master/core-services/image-mirroring/knative/mapping_knative_v1_2_quay
   base="quay.io/openshift-knative/"
   image_template=$(
     cat <<-EOF
 $base{{- with .Name }}
 {{- if eq .      "wathola-kafka-sender"}}{{.}}:v$eventing_kafka_image_version
-{{- else if eq . "kafka-consumer"      }}knative-eventing-kafka-broker-test-kafka-consumer:knative-v$eventing_kafka_broker_image_version
+{{- else if eq . "kafka-consumer"      }}knative-eventing-kafka-broker-test-kafka-consumer:knative-v$eventing_kafka_broker_test_images_version
```
Comment (pierDipi) [inline test/lib.bash:201]: What's the output of this if it is a commit?

This image wouldn't resolve properly:
```
knative-eventing-kafka-broker-test-kafka-consumer:knative-v$eventing_kafka_broker_test_images_version
```

File: olm-catalog/serverless-operator/project.yaml [hunk 2]
```diff
@@ -39,10 +39,17 @@ dependencies:
...
+  # use a commit hash like f24550ef10c9deea57c2306db05d813406ee73b8 (at least 10 chars) if you would like to make a SO
+  #   release with fixed EKB image versions.
+  eventing_kafka_broker: f24550ef10c9deea57c2306db05d813406ee73b8 # --> release v1.4
   # eventing-kafka-broker midstream branch name
-  eventing_kafka_broker_artifacts_branch: release-v1.4
+  eventing_kafka_broker_artifacts_branch: f24550ef10c9deea57c2306db05d813406ee73b8
+  # test images version for eventing-kafka-broker.
+  # as we don't publish these images by commit tag, we need to still reference them using the image version.
+  # this is fine, since the floating tags are overwritten with new images.
+  eventing_kafka_broker_test_images: 1.4
```
Comment (pierDipi) [inline olm-catalog/serverless-operator/project.yaml:52]: Ah, we're introducing yet another version

Comment (pierDipi) [reply olm-catalog/serverless-operator/project.yaml:52]: I guess once we move to "everything to quay" we can get rid of the additional version, right?

Comment (aliok) [reply olm-catalog/serverless-operator/project.yaml:52]: There would be some scripting changes required but yes

**Comment by aliok** [comment]
/close

I am gonna just extract code from https://github.com/openshift-knative/serverless-operator/pull/1792.

Closing in favor of https://github.com/openshift-knative/serverless-operator/pull/1831 and https://github.com/openshift-knative/serverless-operator/pull/1832

**Comment by aliok** [comment]
/reopen

I think this is a better way to go, compared to https://github.com/openshift-knative/serverless-operator/pull/1832

**Comment by aliok** [comment]
/close
Gonna work on this after https://github.com/openshift-knative/serverless-operator/pull/1792

**Pull request closed without merging**